src/app/admin/collectionCentresTable/CollectionCentresTable.tsx

```diff
@@ -1,116 +1,324 @@
+            setErrorMessage("Error fetching data, please reload");
             return;
         }
-
-        setCollectionCentres(data);
+        setRows(data);
+        setIsLoading(false);
     }, []);
 
     useEffect(() => {
-        void fetchAndDisplayCollectionCentres();
-    }, [fetchAndDisplayCollectionCentres]);
+        void getCollectionCentresForTable();
+    }, [getCollectionCentresForTable]);
 
     useEffect(() => {
         const subscriptionChannel = supabase
-            .channel("collection-centres-table-changes")
+            .channel("collection-centre-table-changes")
             .on(
                 "postgres_changes",
                 { event: "*", schema: "public", table: "collection_centres" },
-                fetchAndDisplayCollectionCentres
+                getCollectionCentresForTable
             )
-            .subscribe((status, err) => {
-                if (subscriptionStatusRequiresErrorMessage(status, err, "collection_centres")) {
+            .subscribe((status, error) => {
+                subscriptionStatusRequiresErrorMessage(status, error, "collection_centres") &&
                     setErrorMessage("Error fetching data, please reload");
-                }
             });
 
         return () => {
             void supabase.removeChannel(subscriptionChannel);
         };
-    }, [fetchAndDisplayCollectionCentres]);
+    }, [getCollectionCentresForTable]);
+
+    const handleSaveClick = (id: GridRowId) => () => {
+        setRowModesModel((currentValue) => ({
+            ...currentValue,
+            [id]: { mode: GridRowModes.View },
+        }));
+    };
+
+    const processRowUpdate = async (
+        newRow: CollectionCentresTableRow
+    ): Promise<CollectionCentresTableRow> => {
+        setErrorMessage(null);
+        setIsLoading(true);
+
+        if (newRow.isNew) {
+            const { data: createdCollectionCentre, error: insertCollectionCentreError } =
+                await insertNewCollectionCentre(newRow);
+            const baseAuditLog = getBaseAuditLogForCollectionCentreAction(
+                "add a new collection centre",
+                newRow,
+                { excludeCollectionCentreId: true }
+            );
+
+            if (insertCollectionCentreError) {
+                setErrorMessage(
+                    `Failed to add the collection centre. Log ID: ${insertCollectionCentreError.logId}`
+                );
+                void sendAuditLog({
+                    ...baseAuditLog,
+                    wasSuccess: false,
+                    logId: insertCollectionCentreError.logId,
+                });
+            } else {
+                void sendAuditLog({
+                    ...baseAuditLog,
+                    collectionCentreId: createdCollectionCentre.collectionCentreId,
+                    wasSuccess: true,
+                });
+                setIsLoading(false);
+            }
```

Comment:
can we take this whole bit inside the if (and the bit inside the else) into separate functions and call them `addNewCollectionCentre` and `updateCollectionCentre` or something for readability?